python/pyspark/sql/tests/test_udf.py

```diff
@@ -935,7 +940,8 @@ def test_udf(a, b):
             ]
         ):
             with self.subTest(query_no=i):
-                assertDataFrameEqual(df, [Row(0), Row(101)])
+                if have_pyarrow:
+                    assertDataFrameEqual(df, [Row(0), Row(101)])
```

Comment:
why do we need arrow for compating DataFrame and Rows @itholic ?

Reply:
because `assertDataFrameEqual` try importing pyspark.pandas which requires `arrow` and `pandas`

Reply:
Can we avoid it since we don't need it when we compare PySpark DataFrame and PySpark Rows?

Reply:
On second thought, we can just remove `pyspark.pandas` dependency from `assertDataFrameEqual`.

Just submitted PR: https://github.com/apache/spark/pull/44899.

Reply:
May I confirm why we need `pyarrow`?

```py
>>> import pyarrow
...
ModuleNotFoundError: No module named 'pyarrow'
>>> df = spark.createDataFrame(data=[("1",), ("2",)])
>>> assertDataFrameEqual(df, [Row(_1="1"), Row(_1="2")])
>>> 
```